data/default.nix

```diff
@@ -6,27 +6,47 @@ with (import nixpkgs config);
 
 let
   mkDerivation =
-    { srcs ? ./elm-srcs.nix
-    , src
-    , name
+    { elmSrcs ? ./elm-srcs.nix
+    , srcs
     , srcdir ? "./src"
+    , extension ? ".elm"
+    , name
     , targets ? []
     , versionsDat ? ./versions.dat
     }:
     stdenv.mkDerivation {
-      inherit name src;
+      inherit name srcs;
+      sourceRoot = ".";
 
       buildInputs = [ elmPackages.elm ];
 
       buildPhase = pkgs.elmPackages.fetchElmDeps {
-        elmPackages = import srcs;
+        elmPackages = import elmSrcs;
         inherit versionsDat;
       };
 
       installPhase = let
-        elmfile = module: "\${srcdir}/\${builtins.replaceStrings ["."] ["/"] module}.elm";
+        elmfile = module: "\${srcdir}/\${builtins.replaceStrings ["."] ["/"] module}\${extension}";
+        # TODO: review naming
+        elmJson = pkgs.lib.importJSON ./elm.json;
+        elmJsonFile = pkgs.writeText "elm.json" (builtins.toJSON generatedElmJson);
+        # TODO: hacky.. there might be a better way
+        genSrcs = xs: map (path: if path == "." then srcdir else builtins.replaceStrings [".." "/"] ["" ""] path) xs;
```

Comment:
this is temp implementation.

works: `["../../foo", "../bar", "." ]`
should not work `"foo/bar"`

ideally, we would use same function nix uses for resolving `srcs`. I'm not sure what happens in case of conflict like `["../foo", "./foo"]` yet.